crates/starknet/src/starknet/dump.rs

```diff
@@ -0,0 +1,178 @@
+                        &tx.constructor_calldata,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        tx.class_hash,
+                        tx.contract_address_salt,
+                        tx.version,
+                    );
+                    self.add_deploy_account_transaction(deploy_account_tx)?;
+                }
+                Transaction::Deploy(_) => return Err(Error::SerializationNotSupported),
+                Transaction::Invoke(InvokeTransaction::Version0(_)) => {
+                    return Err(Error::SerializationNotSupported);
+                }
+                Transaction::Invoke(InvokeTransaction::Version1(tx)) => {
+                    let invoke_tx = BroadcastedInvokeTransaction::new(
+                        tx.sender_address,
+                        tx.max_fee,
+                        &tx.signature,
+                        tx.nonce,
+                        &tx.calldata,
+                        tx.version,
+                    );
+                    self.add_invoke_transaction(invoke_tx)?;
+                }
+                Transaction::L1Handler(_) => return Err(Error::SerializationNotSupported),
+            };
+        }
+
+        Ok(())
+    }
+
+    /// attach starknet transaction to end of existing file
+    pub fn dump_transaction(&self, transaction: &Transaction) -> DevnetResult<()> {
+        match &self.config.dump_path {
+            Some(path) => {
+                let file_path = Path::new(path);
+                if file_path.exists() {
+                    // attach to file
+                    let transaction_dump = serde_json::to_string(transaction).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    let mut file = OpenOptions::new()
+                        .append(true)
+                        .write(true)
+                        .open(file_path)
+                        .map_err(Error::IoError)?;
+                    let lenght = file.seek(SeekFrom::End(0)).map_err(Error::IoError)?;
+                    file.set_len(lenght - 1).map_err(Error::IoError)?; // remove last "]" with set_len
+                    file.write_all(format!(", {transaction_dump}]").as_bytes())
+                        .map_err(Error::IoError)?;
+                } else {
+                    // create file
+                    let transactions = vec![transaction];
+                    let transactions_dump = serde_json::to_string(&transactions).map_err(|_| {
+                        Error::SerializationError { obj_name: "Vec<Transaction>".to_string() }
+                    })?;
+                    fs::write(Path::new(&path), transactions_dump)?;
+                }
+
+                Ok(())
+            }
+            None => Err(Error::FormatError),
```

Comment:
I see that here and in two other places, we just return this FormatError. Do we ever expect it to be thrown? If so, there should be some other info inside to help us debug.

Reply:
If CLI argument parsing is done properly, we don't expect this logic branch to ever be reached (the compiler requires it though). Perhaps if #194 will be dealing with this logic, there we can throw a more meaningful error (something that doesn't conceal the real cause and shows both the user and us where the error is coming from).

Marking this as resolved.